src/main/webapp/app/exercises/programming/shared/code-editor/ace/code-editor-ace.component.ts

```diff
@@ -178,38 +181,39 @@ export class CodeEditorAceComponent implements AfterViewInit, OnChanges, OnDestr
             if ((this.selectedFile && !this.fileSession[this.selectedFile]) || this.fileSession[this.selectedFile].loadingError) {
                 this.loadFile(this.selectedFile);
             } else {
-                this.initEditorAfterFileChange();
+                this.initEditor();
             }
         }
         if (changes.commitState && changes.commitState.currentValue === CommitState.CONFLICT) {
             this.editor.setReadOnly(true);
         } else if (changes.commitState && changes.commitState.previousValue === CommitState.CONFLICT && changes.commitState.currentValue !== CommitState.CONFLICT) {
             this.editor.setReadOnly(false);
         }
+        if (changes.feedback || changes.feedbackSuggestions) {
+            // on changes to these props
+            if (this.editorSession) {
+                // initEditor should already be called, otherwise there is no need to init this anyways
+                this.updateLineWidgets();
+            }
+        }
     }
 
     /**
-     * Setup the ace editor after a file change occurred.
-     * Makes sure previous settings are restored and the correct language service is used.
+     * Set up the ace editor after a file change occurred or the file finished loading.
+     * Makes sure potential previous settings are restored and the correct language service is used.
      **/
-    initEditorAfterFileChange() {
-        // Setup editorSession for inline feedback using lineWidgets
+    initEditor() {
```

Comment:
Note for code reviewers: The old name did not make sense: The function is also called initially, not only on file change.